concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/VegasLimit.java

```diff
@@ -183,23 +182,29 @@ private VegasLimit(Builder builder) {
         this.thresholdFunc = builder.thresholdFunc;
         this.smoothing = builder.smoothing;
         this.probeMultiplier = builder.probeMultiplier;
-        this.probeCountdown = nextProbeCountdown();
-        
+
+        resetProbeJitter();
+
         this.rttSampleListener = builder.registry.registerDistribution(MetricIds.MIN_RTT_NAME);
     }
-    
-    private int nextProbeCountdown() {
-        int max = (int) (probeMultiplier * estimatedLimit);
-        return ThreadLocalRandom.current().nextInt(max / 2, max);
+
+    private void resetProbeJitter() {
+        probeJitter = ThreadLocalRandom.current().nextDouble(0.5, 1);
+    }
+
+    private boolean shouldProbe() {
+        return probeJitter * probeMultiplier * estimatedLimit <= probeCount;
```

Comment:
This calculation would make shouldProbe() very dynamic as it is done for each update instead of once after each probing event.  This could result in excessive probing when the limit is low.  Curious to know if you've tried this under light and heavy load to see if this causes any issues.

Reply:
I haven't tried this in prod; it just seemed like a reasonable idea. To me it seemed reasonable because if I have a consistently low limit, I will probe every X time, and if I have a consistently high limit, I will probe every Y time, and so all this should change is the behaviour at the boundary; e.g. if I move from a high limit to a low limit I will switch to probing every X time immediately rather than delayed until the end of the current probing window, and the other way around I will switch to probing every Y time immediately rather than delayed by the end of the current probing window.

I confess I'm not super super clued in on the theory.